ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -619,4 +623,15 @@ static void maybeCreateProcessingLogStream(
 
     commandQueue.enqueueCommand(configured.get());
   }
+
+  private static void maybeConfigureRocksDBConfigSetter(final KsqlConfig ksqlConfig) {
+    final Map<String, Object> streamsProps = ksqlConfig.getKsqlStreamConfigProps();
+    final Class<?> clazz =
+        (Class) streamsProps.get(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG);
+
+    if (clazz != null && org.apache.kafka.common.Configurable.class.isAssignableFrom(clazz)) {
+      ((org.apache.kafka.common.Configurable) Utils.newInstance(clazz))
```

Comment:
`Utils.newInstance()` seems to require that the configurable class be public, as `KsqlRestApplicationTest#shouldConfigureRocksDBConfigSetter()` fails if `ConfigurableTestRocksDBConfigSetter` is not public. Is this a reasonable requirement or have I gone awry with this approach?

If we stick with this approach I'll need to update the error handling around what happens if a non-accessible, configurable, RocksDBConfigSetter class is passed as the current failure is not user-friendly.

Reply:
It's acceptable/normal to require a plugin to have a public no-args constructor.  But we should ensure good error messages if it doesn't have one or if the constructor throws.